packages/Php70/src/Rector/FuncCall/NonVariableToVariableOnFunctionCallRector.php

```diff
@@ -0,0 +1,204 @@
+use PhpParser\Node\Expr\StaticPropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+use PHPStan\Broker\FunctionNotFoundException;
+use PHPStan\Reflection\ParameterReflection;
+use PHPStan\Reflection\ParametersAcceptor;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * @see https://www.php.net/manual/en/migration70.incompatible.php
+ *
+ * @see \Rector\Php70\Tests\Rector\FuncCall\NonVariableToVariableOnFunctionCallRector\NonVariableToVariableOnFunctionCallRectorTest
+ */
+final class NonVariableToVariableOnFunctionCallRector extends AbstractRector
+{
+    /**
+     * @var Broker
+     */
+    private $broker;
+
+    public function __construct(Broker $broker)
+    {
+        $this->broker = $broker;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('TO DO', [new CodeSample('reset(a());', '$a = a(); reset($a);')]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class, Node\Expr\MethodCall::class, Node\Expr\StaticCall::class];
+    }
+
+    /**
+     * @param FuncCall|Node\Expr\MethodCall|Node\Expr\StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $arguments = $this->getNonVariableArguments($node);
+
+        if ($arguments === []) {
+            return null;
+        }
+
+        foreach ($arguments as $key => $argument) {
+            $variable = null;
+            $assign = null;
+
+            if (
+                $argument instanceof Node\Expr\Assign
+                || $argument instanceof Node\Expr\AssignRef
+                || $argument instanceof Node\Expr\AssignOp
+            ) {
+                $variable = $argument->var;
+                $assign = $argument;
+            }
+
+            if (
+                $argument instanceof FuncCall
+                && $argument->name instanceof Node\Name
+            ) {
+                $name = $this->getName($argument);
+
+                if ($name === null) {
+                    continue;
+                }
+
+                $variable = new Variable($name);
+                $assign = new Node\Expr\Assign($variable, $argument);
+            }
+
+            if (
+                $argument instanceof Node\Expr\New_
+            ) {
+                if ($argument->class instanceof Node\Name) {
+                    $variable = new Variable(lcfirst($argument->class->getLast()));
+                } elseif ($argument->class instanceof Node\Stmt\Class_ && $argument->class->name !== null) {
+                    $variable = new Variable(lcfirst($argument->class->name->toString()));
+                } else {
+                    continue;
+                }
```

Comment:
Here you can use:

```php
$this->getName($argument->class)
```

Reply:
In case of a namespaced class, I'll get the FQCN of the class (for example `App\Dummy`), which is not suitable for a variable name. What should I do in this case ?

Reply:
You can use `Rector\CodingStyle\Naming\ClassNaming` service and `getShortName()` method.

It's conventional to process any FQN class/interface/type etc. names